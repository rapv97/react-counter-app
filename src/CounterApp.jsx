import { useState } from 'react';
import PropTypes from 'prop-types';


// const handleAdd = (event) => {
//     console.log(event);
// }

export const CounterApp = ({ value })=>  {
// el useState usa un valor por defecto
const [counter, setCounter] = useState( value );

const agregarValor = ( ) => {
    // forma 1 de cambiar el counter
    // setCounter( counter + 1)

    // forma 2 de cambiar el counter
    setCounter( ( c ) => c + 1);
}

const quitarValor = () => setCounter( counter -1);

const resetValor = () => setCounter( value );

return (
    <>
        <h1> CounterApp </h1>
        <h2> { counter } </h2>

        <button onClick={ agregarValor }>+1</button>
        <button onClick={ quitarValor }>-1</button>
        <button aria-label='btn-reset' onClick={ resetValor }>Reset</button>

    </>
)

}

CounterApp.prototype = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 0
}


