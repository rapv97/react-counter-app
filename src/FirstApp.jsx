import  PropTypes  from "prop-types";
// const newMessage = {
//     message: 'Hola mundo',
//     tittle: 'Richarson'
// }

// const retornar = ( a , b) => {
//     return a + b
// }
// {JSON.stringify(newMessage)}

//desestructuramos las props y recibimos el nombre de la variable que nos manda
// el componente padre

export const FirstApp = ({ title, subTitle, name }) => {

  // const {} = props;
  // console.log(props);
    // const neMessage = 'Richarson';
  return (
    //es sinonimo de fregmento los simbolos <> </> y de esa manera ya no
    //debemos importar a Fragment y podemos regresar un grupo en el return
    <>
        <h1 data-testid="test-title"> { title } </h1>
        <p>{ subTitle  }</p>
        <p>{ subTitle  }</p>
        <p>{ name  }</p>
    </>
    
  )
}

FirstApp.propTypes= {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string
}

FirstApp.defaultProps = {
  // title: 'No hay titulo',
  subTitle: 'No hay subtitulo',
  name: 'Richarson',

}