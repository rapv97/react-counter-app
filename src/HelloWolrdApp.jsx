
// export function App(){

//     return( <h1>Hola Mundo</h1>);
// }

// import React from 'react'

export const HelloWolrdApp = () => {
  return (
    <h1>Hello Wolrd App</h1>
  )
}
