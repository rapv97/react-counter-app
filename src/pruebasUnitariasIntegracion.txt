pruebas unitarias


unitarias y de integracion

Unitarias: Enfocadas en peque√±as funcionalidades

ejemplo la llanta de un vehiculo corroboramos que gire bien, que no hayan
deformaciones etc


Integracion_: enfocadas en como reaccionan varias piezas en conjunto

ejemplo de integracion es probar la llantas integrandolas con el vehiculo
y ver si trabajan correctamente

caracteristicas de las pruebas
1 faciles de escribir
2 faciles de leer
3 confiables
4 rapidas
5 principalmente unitarias

A ARRANGE (ARREGLAR):

paso en el cual nosotros establecemos el paso inicial. se prepara el ambiente a probar
se inicializan variables, se hacen importaciones necesarias etc.

A ACT (ACTUAR):

acciones o estimulos llamar metodos, simular clicks, realizar acciones sobre el paso anterios, aplicamos el estimulo al sujeto de pruebas para verificar su comportamiento


A ASSERT (AFIRMAR)

se resume a observar el comportamiento resultante, que algo cambie, que algo incremente, o que no haya pasado nada




