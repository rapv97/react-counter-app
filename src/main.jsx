import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import { HelloWolrdApp } from './HelloWolrdApp';
import './styles.css'


//todo inicia con un functional component, componente basados en funciones



//desde function App es mi aplicacion ahora hay que renderizarla

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <HelloWolrdApp/> */}
        {/* <FirstApp title='Hola, soy vegeta' /> */}
        <CounterApp value = {123}/>
    </React.StrictMode>
);

