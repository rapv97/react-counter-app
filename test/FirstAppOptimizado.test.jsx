import { render , screen} from "@testing-library/react" 
import { FirstApp } from "../src/FirstApp"

//con el screen obtenemos el objeto que nosotros estamos renderizando
describe('pruebas en <FirstApp />', () => { 
  const title = 'Hola, soy goku';
  const subTitle = 'soy subtitulo';

    test('debe hacer match con el snapshot', () => { 
        const  {container} =  render( <FirstApp title = { title } /> ) 

        //el snapshot hace como una captura de lo que devuelve y lo compara con lo  que deberia devolver
        //en este caso FirstApp se debe ocupar de preferencia hasta que ya se haya culminado el desarrollo
        expect(container).toMatchSnapshot();

     });

     test('debe mostrar el mensaje Hola, soy goku', () => { 
        render(<FirstApp title = { title } />)
        expect(screen.getByText(title)).toBeTruthy();


      })

     test('debe de mostrar el titulo en un h1', () => { 
        render( <FirstApp title = { title } /> ) 

        expect( screen.getByRole('heading', {level: 1}).innerHTML ).toContain( title );

        //buscamos que exista una etiqueta h1
        //y que el texto que contenga sea identico con la const title
        // const h1 = container.querySelector('h1');
        // expect(h1.innerHTML).toBe(title);
        // expect(h1.innerHTML).toContain(title);
      })

      test('debe de mostrar el subtitulo enviado por props', () => { 
               
        render(
           <FirstApp 
              title = { title }
              subTitle= {subTitle}
            />
            ); 
        expect( screen.getAllByText(subTitle).length ).toBe(2);

       })
 });