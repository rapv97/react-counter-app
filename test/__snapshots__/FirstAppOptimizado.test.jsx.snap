// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <FirstApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy goku
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Richarson
  </p>
</div>
`;
